testing/src/main/java/io/grpc/testing/TestUtils.java

```diff
@@ -255,19 +253,20 @@ public static InetSocketAddress testServerAddress(int port) {
    * @param name  name of a file in src/main/resources/certs.
    */
   public static File loadCert(String name) throws IOException {
-    InputStream in = TestUtils.class.getResourceAsStream("/certs/" + name);
+    InputStream in = new BufferedInputStream(TestUtils.class.getResourceAsStream("/certs/" + name));
     File tmpFile = File.createTempFile(name, "");
     tmpFile.deleteOnExit();
 
-    BufferedWriter writer =
-        new BufferedWriter(new OutputStreamWriter(new FileOutputStream(tmpFile), UTF_8));
+    OutputStream os = new BufferedOutputStream(new FileOutputStream(tmpFile));
     try {
       int b;
       while ((b = in.read()) != -1) {
```

Comment:
If we read chunks at a time, we wouldn't actually need the **Buffered**Input/OutputStream. The Buffered streams keep a buffer internally and do reads and writes of chunks at a time.

We read and write a single byte at a time because it is dead-simple and we don't care about performance here.